File: ecommerce-frontend/src/app/services/payment.service.ts
// services/payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PaymentRequestDTO, PaymentResponseDTO } from '../models/payment.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {
  private apiUrl = environment.api.payment;

  constructor(private http: HttpClient) { }

  processPayment(paymentRequest: PaymentRequestDTO): Observable<PaymentResponseDTO> {
    return this.http.post<PaymentResponseDTO>(`${this.apiUrl}/process`, paymentRequest);
  }

  getPaymentById(paymentId: number): Observable<PaymentResponseDTO> {
    return this.http.get<PaymentResponseDTO>(`${this.apiUrl}/${paymentId}`);
  }

  getPaymentByOrderId(orderId: number): Observable<PaymentResponseDTO> {
    return this.http.get<PaymentResponseDTO>(`${this.apiUrl}/order/${orderId}`);
  }

  healthCheck(): Observable<string> {
    return this.http.get<string>(`${this.apiUrl}/health`);
  }

  testRabbitMQ(): Observable<string> {
    return this.http.get<string>(`${this.apiUrl}/test-rabbitmq`);
  }
}

File: payment-service/src/main/java/com/payment/payment_service/service/PaymentService.java
package com.payment.payment_service.service;

import com.payment.payment_service.entity.PaymentRequestDTO;
import com.payment.payment_service.entity.PaymentResponseDTO;

public interface PaymentService {
	/**
     * Process payment for an order
     * @param paymentRequest the payment request containing order details, amount, etc.
     * @return PaymentResponseDTO containing payment processing result
     */
    PaymentResponseDTO processPayment(PaymentRequestDTO paymentRequest);

    /**
     * Retrieve payment details by payment ID
     * @param paymentId the unique identifier of the payment
     * @return PaymentResponseDTO containing payment details
     */
    PaymentResponseDTO getPaymentById(Long paymentId);

    /**
     * Retrieve payment details by order ID
     * @param orderId the unique identifier of the order
     * @return PaymentResponseDTO containing payment details
     */
    PaymentResponseDTO getPaymentByOrderId(Long orderId);

}
